### PR TITLE
PCRM-6994 - prevent files from attempting multipart upload when only one chunk

### PR DESCRIPTION
**Pull request by antonyibbotson1-nhs**

### CODE REVIEW

File: mesh_aws_client/mesh_fetch_message_chunk_application.py [hunk 1]
```diff
@@ -83,59 +166,49 @@ def _get_aws_bucket_and_key(self):
...
                 )
+
             self.s3_client.delete_object(
                 Bucket=self.s3_bucket,
-                Key=s3_tempfile_key,
+                Key=self.s3_tempfile_key,
                 BypassGovernanceRetention=True,
             )
         except ClientError:
-            # Not found
```
Comment (Hobdell) [inline mesh_aws_client/mesh_fetch_message_chunk_application.py:119]: Add warning log ref  here to explain flow

File: mesh_aws_client/mesh_fetch_message_chunk_application.py [hunk 2]
```diff
@@ -36,34 +36,117 @@ def __init__(self, additional_log_config=None, load_ssm_params=False):
...
+        )
+
+    def _handle_report_message(self):
+        self.log_object.write_log(
+            "MESHFETCH0010", None, {"message_id": self.message_id}
+        )
+        buffer = json.dumps(dict(self.http_response.headers)).encode("utf-8")
+        self.http_headers_bytes_read = len(buffer)
+        self._upload_to_s3(buffer, s3_key=self.s3_key)
+        self.mailbox.acknowledge_message(self.message_id)
```
Comment (Hobdell) [inline mesh_aws_client/mesh_fetch_message_chunk_application.py:148]: log message to say it's uploaded and acknowledged?

Comment (Hobdell) [reply mesh_aws_client/mesh_fetch_message_chunk_application.py:148]: k, there's a log ref when function _upload_to_s3  called 👍🏻

File: mesh_aws_client/mesh_fetch_message_chunk_application.py (hunk 1 above)
Comment (antonyibbotson1-nhs) [reply mesh_aws_client/mesh_fetch_message_chunk_application.py:119]: Added as an INFO log, this occurs when there it can't find an overflow file

**Review by Hobdell — DISMISSED** [review]

**Review by Hobdell — APPROVED** [review]

**Pull request merged**